specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-06-01-preview/SourceControls.json

```diff
@@ -404,8 +417,41 @@
       ],
       "type": "object"
     },
+    "RepositoryAccess": {
+      "type": "object",
+      "properties": {
+        "kind": {
+          "description": "The kind of repository access credentials",
+          "$ref": "#/definitions/RepositoryAccessKind"
+        },
+        "code": {
+          "type": "string",
+          "description": "OAuth Code. Required when `kind` is `OAuth`"
+        },
+        "state": {
+          "type": "string",
+          "description": "OAuth State. Required when `kind` is `OAuth`"
+        },
+        "clientId": {
+          "type": "string",
+          "description": "OAuth ClientId. Required when `kind` is `OAuth`"
+        },
+        "token": {
+          "type": "string",
```

Comment:
Use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret) to mark the secret.